_posts/2019-05-21-accessing-kafka-part-5.md

```diff
@@ -0,0 +1,218 @@
+The other parts published so far are:_
+
+* _[Part 1 - Introduction](https://strimzi.io/2019/04/17/accessing-kafka-part-1.html)_
+* _[Part 2 - Node Ports](https://strimzi.io/2019/04/23/accessing-kafka-part-2.html)_
+* _[Part 3 - OpenShift Routes](https://strimzi.io/2019/04/30/accessing-kafka-part-3.html)_
+* _[Part 4 - Load Balancers](https://strimzi.io/2019/05/13/accessing-kafka-part-4.html)_
+* _Part 5 - Ingress (this post)_
+
+_This is the last post in the series._
+
+## Kubernetes Ingress
+
+Ingress is a Kubernetes API for managing external access to HTTP/HTTPS services which was added in Kubernetes 1.1.
+Ingress is the Kubernetes counter part to the OpenShift Routes which we discussed in [part 3](https://strimzi.io/2019/04/30/accessing-kafka-part-3.html).
+It acts as a [Layer 7](https://en.wikipedia.org/wiki/OSI_model#Layer_7:_Application_Layer) load balancer for HTTP or HTTPS traffic.
+The Ingress resources will define the rules for routing the traffic to different services and pods.
+And an Ingress controller will take care of the actual routing.
+For more information about Ingress, check the [Kubernetes website](https://kubernetes.io/docs/concepts/services-networking/ingress/).
+
+Ingress is a bit strange part of the Kubernetes API.
+The Ingress API it self is part of every Kubernetes cluster.
+But the Ingress controller which would do the routing is not part of core Kubernetes.
+So while you will be able to create the Ingress resources, there might be nothing to actually route the traffic.
+
+So for the Ingress resources to actually do something, you need to make sure an Ingress controller is installed.
+There are many different Ingress controllers.
+The Kubernetes project it self has two controllers: 
+* [NGINX controller](https://github.com/kubernetes/ingress-nginx)
+* [GCE controller](https://github.com/kubernetes/ingress-gce) for Google Cloud
+
+But there are many additional controllers created and maintained by different communities and companies.
+A list of different controllers can be found on the [Kubernetes website](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/).
+
+Most of the controllers rely on a load balancer or node port service which will get the external traffic to the controller.
+Once the traffic reaches the controller, the controller will route it based in the rules specified in the Ingress resource to the different services and pods.
+The controller it self usually also runs as yet another application inside the Kubernetes cluster.
+
+![Accessing applications using NGINX Ingress Controller]({{ "/assets/2019-05-21-ingress-controller.png" }})
+
+Some of the controllers are tailored for a specific public clouds - for example the [AWS ALB Ingress Controller](https://github.com/kubernetes-sigs/aws-alb-ingress-controller) is provisioning the AWS Application Load Balancer to do the routing instead of doing it inside some pod in your Kubernetes cluster.
```

Comment:
```suggestion
Some of the controllers are tailored for a specific public cloud - for example the [AWS ALB Ingress Controller](https://github.com/kubernetes-sigs/aws-alb-ingress-controller) provisions the AWS Application Load Balancer to do the routing instead of doing it inside some pod in your Kubernetes cluster.
```